File: repair-management-api/Models/RepairJob.cs
using RepairManagementApi.Enums;

namespace RepairManagementApi.Models;

public class RepairJob
{
    public Guid Id { get; set; }
    public Guid CustomerId { get; set; }
    public Customer? Customer { get; set; }
    public Guid DeviceId { get; set; }
    public Device? Device { get; set; }
    public Guid BranchId { get; set; }
    public Branch? Branch { get; set; }
    public string JobNumber { get; set; } = string.Empty;
    public string ProblemDescription { get; set; } = string.Empty;
    public string? DiagnosisNotes { get; set; }
    public string? ResolutionNotes { get; set; }
    public decimal? EstimatedCost { get; set; }
    public decimal? FinalCost { get; set; }
    public RepairJobStatus Status { get; set; }
    public DateTime ReceivedAtUtc { get; set; }
    public DateTime? CompletedAtUtc { get; set; }
    public DateTime CreatedAtUtc { get; set; }
    public DateTime UpdatedAtUtc { get; set; }
}

File: repair-management-app/src/types/repairJob.ts
export const repairJobStatuses = [
  "Received",
  "Diagnosing",
  "Repairing",
  "ReadyForPickup",
  "Completed",
  "Cancelled",
] as const;

export type RepairJobStatus = (typeof repairJobStatuses)[number];

export type RepairJobListItem = {
  id: string;
  customerId: string;
  deviceId: string;
  jobNumber: string;
  problemDescription: string;
  status: RepairJobStatus;
  estimatedCost: number | null;
  receivedAtUtc: string;
};

export type RepairJobResponse = {
  id: string;
  customerId: string;
  deviceId: string;
  branchId: string;
  jobNumber: string;
  problemDescription: string;
  diagnosisNotes: string | null;
  resolutionNotes: string | null;
  estimatedCost: number | null;
  finalCost: number | null;
  status: RepairJobStatus;
  receivedAtUtc: string;
  completedAtUtc: string | null;
  createdAtUtc: string;
  updatedAtUtc: string;
};

export type UpdateRepairJobRequest = {
  problemDescription: string;
  diagnosisNotes: string | null;
  resolutionNotes: string | null;
  estimatedCost: number | null;
  finalCost: number | null;
};

export type UpdateRepairJobStatusRequest = {
  status: RepairJobStatus;
};
